src/main/scala/org/neo4j/spark/rdd/Neo4jTupleRDD.scala

```diff
@@ -13,23 +16,9 @@ class Neo4jTupleRDD(@transient sc: SparkContext, val query: String, val paramete
   private val config = Neo4jConfig(sc.getConf)
 
   override def compute(split: Partition, context: TaskContext): Iterator[Seq[(String, AnyRef)]] = {
-    val driver: Driver = config.driver()
-    val session = driver.session()
-    try {
-      val result = session.run(query, parameters.toMap.asJava)
-
-      result.asScala.map((record) => {
-        val res = record.asMap().asScala.toSeq
-        if (!result.hasNext) {
-          if (session.isOpen) session.close()
-          driver.close()
-        }
-        res
-      })
-    } finally {
-      if (session.isOpen) session.close()
-      driver.close()
-    }
+    new Neo4jSessionAwareIterator(config, query, parameters.toMap.asJava, false).map(record => {
+      record.asMap().asScala.toSeq
```

Comment:
do we have to convert to Spark types here? Perhaps separate PR/issue

Reply:
Done in this one